src/runtime/virtcontainers/mount.go

```diff
@@ -206,24 +206,40 @@ func getDeviceForPath(path string) (device, error) {
 	return dev, nil
 }
 
-var blockFormatTemplate = "/sys/dev/block/%d:%d/dm"
+var checkStorageDriver = checkBlockDevice
 
-var checkStorageDriver = isDeviceMapper
-
-// isDeviceMapper checks if the device with the major and minor numbers is a devicemapper block device
-func isDeviceMapper(major, minor int) (bool, error) {
+var blockFormatTemplates = map[string]string{
+	"dm":    "/sys/dev/block/%d:%d/dm",
+	"block": "/sys/dev/block/%d:%d/device/block",
+	"loop":  "/sys/dev/block/%d:%d/loop",
+}
 
-	//Check if /sys/dev/block/${major}-${minor}/dm exists
-	sysPath := fmt.Sprintf(blockFormatTemplate, major, minor)
+func isBlockDevice(major, minor int, formatTemplate string) (bool, error) {
+	sysPath := fmt.Sprintf(formatTemplate, major, minor)
 
 	_, err := os.Stat(sysPath)
 	if err == nil {
 		return true, nil
 	} else if os.IsNotExist(err) {
 		return false, nil
+	} else {
+		return false, err
+	}
+}
+
+func checkBlockDevice(major, minor int) (bool, error) {
+	for dev, template := range blockFormatTemplates {
+		isBD, err := isBlockDevice(major, minor, template)
+		if err != nil {
+			return false, err
```

Comment:
Can we add UTs to cover all the branches for the newly added code?

Reply:
The unchecked branch is `err != nil` which only happens when `isBlockDevice` return error. That happens when `os.Stat` fails and not because of `os.IsNotExist`. 
I really don't know when it happens. Do you have any idea?

Reply:
I think the easiest way to accomplish this here is to use a path with an invalid character (the null character `\000` will work unix and windows):
```
fmt.Println("Example: Invalid character")
_, err := os.Stat("\000path")
fmt.Println(err)
```

You can override the `blockFormatTemplates` like you did the tests below to get a path with `\000` in it

(Permissions errors should also work, but might be a little hard to setup test for)

Also, if you need help getting this PR to the finish line, please let me know

Reply:
@sazzy4o I have supplemented the test as you suggested.
It's really grateful that you can help finishing this PR.